front/components/workspace/ProviderManagementModal.tsx

```diff
@@ -0,0 +1,136 @@
+import { ContextItem, Modal, SliderToggle } from "@dust-tt/sparkle";
+import type { ModelProviderIdType, WorkspaceType } from "@dust-tt/types";
+import { SUPPORTED_MODEL_CONFIGS } from "@dust-tt/types";
+import { MODEL_PROVIDER_IDS } from "@dust-tt/types";
+import _ from "lodash";
+import { useCallback, useMemo, useState } from "react";
+
+import { MODEL_PROVIDER_LOGOS } from "@app/components/assistant_builder/InstructionScreen";
+
+interface ModelManagementModalProps {
+  owner: WorkspaceType;
+  showProviderModal: boolean;
+  onClose: () => void;
+  onSave: (updateProviders: ModelProviderIdType[]) => void;
+}
+
+type ProviderStates = Record<ModelProviderIdType, boolean>;
+
+const prettyfiedProviderNames: { [key in ModelProviderIdType]: string } = {
+  openai: "Open AI",
+  anthropic: "Anthropic",
+  mistral: "Mistral AI",
+  google_ai_studio: "Google",
+};
+
+function isLegacyModel(modelName: string): boolean {
+  const legacyPattern = /\d+\.\d+(?!.*latest$)/;
+  return legacyPattern.test(modelName);
+}
```

Comment:
- Anthropic and OpenAI legacy models follow a ` \d+\.\d+` pattern.
- Mistral doesn't have legacy models
- Google neither but Gemini ID  is `1gemini-1.5-pro-latest`

--> looking for any ` \d+\.\d+` pattern that doesn't end with `latest`

Reply:
Not relevant anymore